system/modules/admin/models/Comment.php

```diff
@@ -49,6 +49,18 @@ public function __toString() {
         $str.= "<i>" . formatDateTime($this->dt_created) . "</i>";
         return $str;
     }
+    
+    /*
+     * get object for comment thread
+     * return object
+     */
+    public function getParentObject() {
+        if ($this->obj_table == 'comment') {
+            return $this->w->Comment->getComment($this->obj_id)->getParentObject();
+        } else {
+            return $this->w->Comment->getObject($this->obj_table,$this->obj_id);
```

Comment:
This looks like recursive function but it is calling the same method on the parent object each time. comments can be chained together eg. reply on reply on reply etc. The function keeps going until it finds a non-comment object